### PR TITLE
create JoinNarcolepsyLink page

### PR DESCRIPTION
**Pull request by martindev727**

### CODE REVIEW

File: src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx [hunk 1]
```diff
@@ -1,7 +1,172 @@
...
+import FloatingInputLabel from '../floating-input-label/floating-input-label.component'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { DropdownButton, MenuItem } from 'react-bootstrap'
+import stateList from '../states'
+import './join-narcolepsy-link.component.css'
 
-const JoinNarcolepsyLinkComponent = () => {
-  return <div />
+const specialityList = [
+  {id: 'sleep_specialist', name: 'Sleep Specialist'},
```
Comment (rphovley) [inline src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx]: I just got this from creative for the list of specialties 

Sleep Specialist
Psychiatrist
Neurologist
Pulmonologist
Primary Care Physician
Internal Medicine Physician
Nurse Practitioner/Physician Assistant
Sleep Technologist
Other Healthcare Professional

Comment (martindev727) [reply src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx]: Done

File: src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx [hunk 2]
```diff
@@ -1,7 +1,172 @@
...
+              </div>
+              <div className="form-row">
+                <FloatingInputLabel id="emailAddress" className="input-large" shrink={60} label="Email Address*" onChange={(val) => this.handleInputChange('emailAddress', val)} />
+                <FloatingInputLabel id="degree" className="input-normal" shrink={60} label="Degree" onChange={(val) => this.handleInputChange('degree', val)} />
+              </div>
+              <div className="form-row">
+                <FloatingInputLabel id="address" className="input-normal" shrink={60} label="Address*" onChange={(val) => this.handleInputChange('address', val)} />
+                <FloatingInputLabel id="city" className="input-normal" shrink={60} label="City*" onChange={(val) => this.handleInputChange('city', val)} />
+                <div className="react-floating-label-input input-small">
+                  <select
```
[the remark this answers is not in the document]
Comment (martindev727) [reply src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx]: Fixed

File: src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx [hunk 3]
```diff
@@ -1,7 +1,172 @@
...
+            <p className="form-comment">
+              Please provide the following information to stay connected with Narcolepsy Link and receive occasional updates about
+              this educational initiative.
+            </p>
+          </div>
+          <div className="page-container">
+            <p className="form-header">Enter your information</p>
+            <form>
+              <div className="form-row">
+                <FloatingInputLabel id="firstName" className="input-normal" shrink={60} label="First Name*" onChange={(val) => this.handleInputChange('firstName', val)} />
```
[the remark this answers is not in the document]
Comment (martindev727) [reply src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx]: Fixed

[the remark this answers is not in the document]
Comment (rphovley) [reply src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx]: screen shot 2018-08-29 at 9 02 44 am

On mobile some of the bottom lines on the input are hidden. Maybe use padding-bottom: 21px instead so they are all visible

File: src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx [hunk 4]
```diff
@@ -1,7 +1,180 @@
...
+                    )}
+                  </select>
+                </div>
+              </div>
+              <div className="form-row">
+                <FloatingInputLabel id="national_provider" className="input-normal" shrink={60} label="National Provider Identifier*" onChange={(val) => this.handleInputChange('nationalProvider', val)} />
+              </div>
+              <div className="form-row checkbox-group">
+                <div className="checkbox-item">
+                  <input type="checkbox" id="join_check" name="JoinCheck" required />
```
Comment (rphovley) [inline src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx]: screen shot 2018-08-29 at 9 03 14 am

Checkboxes don't match designs

File: src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (martindev727) [reply src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx]: Fixed

File: src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (martindev727) [reply src/components/join-narcolepsy-link/join-narcolepsy-link.component.jsx]: Fixed

**Comment by martindev727** [comment]
@rphovley  Fixed merge conflicts.

**Pull request closed without merging**